### ISSUE TITLE
Translations doesn't work when switching language with disabled useCookie property

### ISSUE THREAD

**Issue by thnima** [nuxt-community/nuxt-i18n#563]
### Version

[v6.4.1](https://github.com/nuxt-community/nuxt-i18n/releases/tag/v6.4.1)

### Reproduction link

[https://codesandbox.io/s/thirsty-lake-m7vwr](https://codesandbox.io/s/thirsty-lake-m7vwr)

### Steps to reproduce

Click on the link with the label: "Deutsch"

### What is expected ?

The string "English string" should get translated to "Deutscher string"

### What is actually happening?

Clicking the link changes the route to "/de", which is fine. But the text doesn't get translated.
If you set `useCookie: true` in the nuxtConfig again, it works as expected.
Same behaviour can be observed, when you set `useCookie: true` AND disable the cookies in the browser

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt-i18n/issues/c370">#c370</a>)</em></sub></div>

**Comment by rchl** [comment]
The module logic should be improved but I think setting `detectBrowserLanguage` to `false` should work without changing the logic.

Currently you have it set to:
```
detectBrowserLanguage: {
          useCookie: false,
          alwaysRedirect: false,
          cookieKey: "i18n_redirected",
          fallbackLocale: "en"
        }
```
which is basically disabled.

**Comment by thnima** [comment]
True, setting it to false works indeed.
Maybe I should have mentioned the actual use case:
The problem here is: I want to use the cookie,  if cookies are enabled. But on environments where cookies are disabled, it doesn't work.
So assume setting "useCookie: true" and disaable your browser cookie. Then the behaviour is the same, as you set "useCookie: false"

I will add this to the description

**Comment by gokhancetiner** [comment]
Hello @rchl,

Do you have a deadline to fix this bug? 

Many Thanks,

**Comment by rchl** [comment]
I'm trying to think of what would be the expected flow in that case...

- Since `detectBrowserLanguage` is set we try to detect browser language and redirect to a matching route.
- Since cookie is not set (or `useCookie` is false) we always behave like it's initial navigation and trigger redirect.

It's not that obvious how this should be fixed...
While there is a hacky way to test *on the client* if cookies are enabled and maybe act differently in that case, there is no way to do that on the server. So I could change the code so that client-side navigations are not affected by this problem but once you reload the page (or even navigate in non-SPA fashion), it will still redirect to the fallback locale. So it would only be half-fixed.

I wonder if anyone has some solid ideas that I might be overlooking?

**Comment by rchl** [comment]
Also, the `useCookie` option is documented with `Set to false to redirect every time` so it's really doing what it's supposed to. If we assume that, the only issue I see is that it is possible to end up in this broken state where locale in the URL changes but not the actual locale.

**Comment by gokhancetiner** [comment]
> obvious

Hello @rchl ,

You are right we can use useCookie is false and it works for redirection but the problem is if we set useCookie is false and if we change the lang, the URL changes but translations don't work. 

If we set useCookie is true I have another problem explained below link

https://cmty.app/nuxt/nuxt-i18n/issues/c397

Many Thanks

**Comment by rchl** [comment]
Yes, but as I've explained, that's pretty much what such configuration is supposed to do - always follow browser language and not allow changes to the locale.

Your other issue with `setCookie = true` looks like #491.

**Comment by gokhancetiner** [comment]
> Yes, but as I've explained, that's pretty much what such configuration is supposed to do - always follow browser language and not allow changes to the locale.
> 
> Your other issue with `setCookie = true` looks like #491.

My problem is if browserLang and cookie lang are different redirections don't work.
On #491 you set default lang but I want to redirect user's default browser lang

**Comment by Pijuli** [comment]
Same issue here.
Temporary fixed it with @rchl T.y.
hope there's not many user with cookies disabled 👼

**Comment by gokhancetiner** [comment]
> Same issue here.
> Temporary fixed it with @rchl T.y.
> hope there's not many users with cookies disabled 👼

Hello @Pijuli ,

What is temporary fixed ?

**Comment by rchl** [comment]
I have made a fix that should address that in #651 (not yet integrated at the time of writing).

**Comment by thnima** [comment]
> I have made a fix that should address that in #651 (not yet integrated at the time of writing).

Amazing, everything works smooth now. Thanks!
